### PR TITLE
feat(wallet): add receive screen on wallet home and make wallet tab f…

### LINKED ISSUE #20220: Wallet - receive page not correct on wallet home
Currently the receive page on the wallet home is broken as it uses the accouht receive screen but this screen depends on an account being selected.

As a temporary quick fix design/product agreed to use the same screen as shell share but with wallet tab displaying first .

### PR DESCRIPTION
**Pull request by J-Son89**
fixes: https://github.com/status-im/status-mobile/issues/20220
This pr makes two small fixes.
1 - Receive option found when a user long presses on an asset in the wallet home page.
Brings the user to the share screen - this is a temporary solution, eventually it will be to a very similar screen but will be closer to the receive page in wallet with multiple options.

2 - The Share screen default tab should show the wallet tab first when the user is on the wallet tab.
Otherwise it will be the profile tab.
I think @qoqobolo suggested this a while back and I couldn't find a relevant issue but thought I would add this here as it was agreed by design etc too 👍 

**Long Press on wallet home and press Receive**

https://github.com/status-im/status-mobile/assets/22799766/5e8092a7-6f5e-4b6a-bfdb-93bb2cd0893f

**Default tab is wallet when the active screen is wallet**
https://github.com/status-im/status-mobile/assets/22799766/08361ad4-2907-4dbb-9b53-99263a0dfa71

### CODE REVIEW

**Review by ajayesivan — APPROVED** [review]

**Review by Parveshdhull — APPROVED** [review]

File: src/status_im/contexts/shell/share/view.cljs [hunk 1]
```diff
@@ -50,7 +51,7 @@
          {:size           28
           :blur?          true
           :on-change      #(reset! selected-tab %)
-          :default-active :profile
+          :default-active initial-tab
```
Comment (OmarBasem) [inline src/status_im/contexts/shell/share/view.cljs]: Should we be refactoring components to a component without a nested render function when we come across them?

**Review by mohsen-ghafouri — APPROVED** [review]

File: src/status_im/contexts/shell/share/view.cljs (hunk 1 above)
Comment (J-Son89) [reply src/status_im/contexts/shell/share/view.cljs]: I think that's on us to decide on how "safe" the refactors are, but leading up to the 2.30 release I think we should keep unnecessary refactors to a minimum unless there is an important reason why, e.g performance issues. 
imo at this point it's advantageous to do whatever we can do to reduce bugs creeping in
cc @cammellos wdyt? 🤔

File: src/status_im/contexts/wallet/common/activity_tab/view.cljs [hunk 2]
```diff
@@ -24,14 +24,14 @@
   (let [chain-id            (or (:chain-id token-in) (:chain-id token-out))
         amount-in-units     (native-module/hex-to-number
                              (utils.hex/normalize-hex amount-in))
-        amount-in-value     (money/with-precision
-                             (money/wei->ether amount-in-units)
-                             precision)
+        amount-in-value     (str (money/with-precision
```
Comment (J-Son89) [inline src/status_im/contexts/wallet/common/activity_tab/view.cljs:27]: this fixes the schema issue on activity tab 👍

**Review by mmilad75 — APPROVED** [review]

**Review by OmarBasem — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hey @J-Son89, thanks for the PR and fixes!
I've got one question/issue so far:

### ISSUE 1: The QR code on the profile screen "remembers" the state of the previously opened QR

So, if you open the `wallet` QR (on the `wallet` screen) first and then go to `profile` and open the QR on the `profile` screen - the **`wallet`** QR will be opened again.
If you go to the `messages`/`communities` screen after that, open QR there, then go to `profile` and open QR on the profile screen - the **`profile`** QR code will be opened.

Is this behavior expected? It looks like we should only show the `Profile QR` on the profile screen, WDYT?🤔

**Steps to reproduce:**
1. Open wallet
2. Press the `QR` icon at the top right (`wallet` QR is shown)
3. Open your profile
4. Press the `QR` icon at the top right (`wallet` QR is shown again)
5. Open the `communities`/`messages` screen
6. Press the `QR` icon at the top right (`profile` QR is shown)
7. Open your profile
8. Press the `QR` icon at the top right (`profile` QR is shown)

https://github.com/status-im/status-mobile/assets/67952253/fdd5f9db-c44b-44d7-9590-7e43b67a1633

---

I also noticed that according to [designs](https://www.figma.com/design/PiN0uwRgZj4L9U4N7xO7Xb/Profile-for-Mobile?node-id=3802-260430&t=TlQTf3T8OfGEB8tB-4) we should not show the `Wallet` tab on the `Share` screen at all if you open it from profile.
But it works like this in develop. Do you know if it's been done on purpose (that we show both `profile` and `wallet` tabs on the Share screen in profile)? 

Design:
Screenshot 2024-05-30 at 14 53 09

Implementation:

Screenshot 2024-05-30 at 14 53 55

**Comment by J-Son89** [comment]
thanks @qoqobolo, going to look into this. I never noticed about the designs for profile but that's a good point. I think it's beyond the scope of this issue but we should raise an issue if it's supposed to be that way. 
However I do think it's strange behaviour since it's the same set of buttons as in the home page tabs etc so probably we should confirm this with the designers first to make sure it's not an old design (pre wallet)

**Comment by qoqobolo** [comment]
> thanks @qoqobolo, going to look into this. I never noticed about the designs for profile but that's a good point. I think it's beyond the scope of this issue but we should raise an issue if it's supposed to be that way. However I do think it's strange behaviour since it's the same set of buttons as in the home page tabs etc so probably we should confirm this with the designers first to make sure it's not an old design (pre wallet)

I'll ask the designers 👍

**Comment by J-Son89** [comment]
thanks for spotting that @qoqobolo - I have addressed that issue now and should show `profile` everywhere else except from wallet home.

**Comment by qoqobolo** [comment]
Everything looks good to me now, thanks for your work @J-Son89 !
PR can be merged.

**Pull request merged**